src/main/java/javax/security/identitystore/CredentialValidationResult.java

```diff
@@ -127,19 +134,20 @@ public CredentialValidationResult(CallerPrincipal callerPrincipal, List<String>
      * @param callerPrincipal Validated caller
      * @param groups Groups associated with the caller from the identity store
      */
-    public CredentialValidationResult(Status status, CallerPrincipal callerPrincipal, List<String> groups) {
+    private CredentialValidationResult(Status status, CallerPrincipal callerPrincipal, List<String> groups) {
 
         if (status == null) {
             throw new NullPointerException("status");
         }
 
         this.status = status;
-        this.callerPrincipal = callerPrincipal;
 
         if (status == VALID) {
+            this.callerPrincipal = callerPrincipal;
             // TODO: to be discussed, should we use null or empty list?
             this.groups = groups != null ? unmodifiableList(new ArrayList<>(groups)) : emptyList();
         } else {
+            this.callerPrincipal = null;
             this.groups = null;
```

Comment:
@arjantijms I noticed you just accepted my PR. Could you please review this point?

Thanks!